wicket-extensions/src/main/java/org/apache/wicket/extensions/ajax/markup/html/AjaxLazyLoadPanel.java

```diff
@@ -187,8 +237,56 @@ public Component getLoadingComponent(final String markupId)
 	{
 		IRequestHandler handler = new ResourceReferenceRequestHandler(
 			AbstractDefaultAjaxBehavior.INDICATOR);
-		return new Label(markupId, "<img alt=\"Loading...\" src=\"" +
-			RequestCycle.get().urlFor(handler) + "\"/>").setEscapeModelStrings(false);
+		return new Label(markupId,
+			"<img alt=\"Loading...\" src=\"" + RequestCycle.get().urlFor(handler) + "\"/>")
+				.setEscapeModelStrings(false);
 	}
 
+	/**
+	 * Factory method for creating the lazily loaded component that replaces the spinner after
+	 * {@link #isReadyForReplacement()} returns {@code true}. You may call setRenderBodyOnly(true)
+	 * on this component if you need the body only.
+	 * 
+	 * @param markupId
+	 *            The components markupid.
+	 * @return The component that must be lazy created.
+	 */
+	public abstract Component getLazyLoadComponent(String markupId);
+
+	/**
+	 * The AJAX timer for updating the AjaxLazyLoadPanel. Is designed to be a page-local singleton
+	 * keeping track of multiple LazyLoadPanels using reference counting.
+	 */
+	private class AjaxLazyLoadTimer extends AbstractAjaxTimerBehavior
+	{
+		private static final long serialVersionUID = 1L;
+
+		private int lazyLoadPanels = 0;
```

Comment:
Counting all loading panels might be problematic: What if one (or one of its parents) is removed from the component hierarchy? The timer will keep requesting until forever.
How about broadcasting the AjaxLazyLoadTimer to all components, giving them the opportunity to keep the timer running?

    ((AjaxLazyLoadTimer)event.getPayload()).notReadyForReplacement();

If any of the panels is not ready yet, the timer has to keep running.